### ISSUE TITLE
Upgrade istio from 1.0.3 to 1.1

### ISSUE THREAD

**Issue by clyang82** [istio/istio#12154]
**Describe the bug**

install istio 1.0.3 with  `helm install install/kubernetes/helm/istio --name istio --namespace istio-system --tls` and perform `kubectl get pod` to verify the installation is success.
```
root@cyang1:~/istio-1.0.3# kubectl get pod
NAME                                     READY     STATUS    RESTARTS   AGE
istio-citadel-78df8c67d9-vvffr           1/1       Running   0          5m
istio-egressgateway-674686c846-mmbhg     1/1       Running   0          5m
istio-egressgateway-674686c846-sw6s9     1/1       Running   0          1m
istio-egressgateway-674686c846-w8rt5     1/1       Running   0          1m
istio-galley-58f566cb66-8nzcl            1/1       Running   0          5m
istio-ingressgateway-6bbdd58f8c-g9crd    1/1       Running   0          5m
istio-ingressgateway-6bbdd58f8c-zf9bq    1/1       Running   0          1m
istio-pilot-7fc859bf4f-6567j             2/2       Running   0          5m
istio-policy-68797d879-db2jl             2/2       Running   0          1m
istio-policy-68797d879-vdvdh             2/2       Running   0          5m
istio-sidecar-injector-b88dfb954-dzhmb   1/1       Running   0          5m
istio-telemetry-68787476f4-ksmbl         2/2       Running   0          3m
istio-telemetry-68787476f4-xwdjh         2/2       Running   0          5m
prometheus-76db5fddd5-fqmwd              1/1       Running   0          5m
```

follow this steps to upgrade to release 1.1 - https://preliminary.istio.io/docs/setup/kubernetes/upgrading-istio/#helm-upgrade
I am using `istio-1.1.0-rc.0` package.

There is an error for this upgrade
```
root@cyang1:~/istio-1.1.0-rc.0# helm upgrade istio install/kubernetes/helm/istio --namespace istio-system --tls
Error: UPGRADE FAILED: no MeshPolicy with the name "default" found
```
Here is content of `default` MeshPolicy

```
apiVersion: v1
items:
- apiVersion: authentication.istio.io/v1alpha1
  kind: MeshPolicy
  metadata:
    annotations:
      kubectl.kubernetes.io/last-applied-configuration: |
        {"apiVersion":"authentication.istio.io/v1alpha1","kind":"MeshPolicy","metadata":{"annotations":{},"labels":{"app":"istio-security","chart":"security-1.0.3","heritage":"Tiller","release":"istio"},"name":"default","namespace":""},"spec":{"peers":[{"mtls":{"mode":"PERMISSIVE"}}]}}
    creationTimestamp: 2019-02-28T13:40:58Z
    generation: 1
    labels:
      app: istio-security
      chart: security-1.0.3
      heritage: Tiller
      release: istio
    name: default
    namespace: ""
    resourceVersion: "8391810"
    selfLink: /apis/authentication.istio.io/v1alpha1/meshpolicies/default
    uid: 7ae4dd5a-3b5e-11e9-882a-00163e019fae
  spec:
    peers:
    - mtls:
        mode: PERMISSIVE
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

**Expected behavior**
1.  there is no error when execute upgrade
2. the `MeshPolicy` chart should be ` chart: security-1.1.0` instead of `chart: security-1.0.3`

cc @linsun @morvencao @gyliu513

**Comment by linsun** [comment]
Could this be caused by the fact meshpolicies.authentication.istio.io is promoted from v1alpha1 in 1.0.x to v1beta1 in 1.1?

cc @wattli @liminw

**Comment by linsun** [comment]
cc @sdake @costinm

**Comment by linsun** [comment]
@clyang82 since we know helm doesn't handle CRDs well during upgrade, can you please use kubectl apply to update the CRDs?   (https://preliminary.istio.io/docs/setup/kubernetes/upgrading-istio/#kubernetes-rolling-update step no. 1) .

If it works,  then we can update this doc: https://preliminary.istio.io/docs/setup/kubernetes/upgrading-istio/#helm-upgrade

**Comment by morvencao** [comment]
@clyang82 What if adding `--force` with `helm upgrade ...`

**Comment by morvencao** [comment]
@clyang82 
After investigation, PR: https://github.com/istio/istio/pull/11418 breaks Istio chart upgrade. 
In 1.0.x, the `meshPolicy` and related `desrinationRule` are created by a helm post-install hook, and all resources deployed by `hook` are not managed by tiller. After the PR: https://github.com/istio/istio/pull/11418, the `meshPolicy` and related `desrinationRule` are create by `helm install`, not the hook, that's why you get `
no MeshPolicy with the name "default" found` while the `meshPolicy` is actually there.

**Comment by morvencao** [comment]
Local test results:
```
[root@master istio-1.0.3]# helm install install/kubernetes/helm/istio --name istio --namespace istio-system
NAME:   istio
LAST DEPLOYED: Thu Feb 28 19:33:28 2019
NAMESPACE: istio-system
STATUS: DEPLOYED
...

[root@master istio-1.0.3]# helm list
NAME 	REVISION	UPDATED                 	STATUS  	CHART      	APP VERSION	NAMESPACE
istio	1       	Thu Feb 28 19:33:28 2019	DEPLOYED	istio-1.0.3	1.0.3      	istio-system
[root@master istio-1.0.3]# kubectl get pod
NAME                                     READY     STATUS    RESTARTS   AGE
istio-citadel-78df8c67d9-9sw88           1/1       Running   0          13m
istio-egressgateway-674686c846-mm5nf     1/1       Running   0          13m
istio-galley-58f566cb66-655m2            1/1       Running   0          13m
istio-ingressgateway-6bbdd58f8c-hk7r7    1/1       Running   0          13m
istio-pilot-7fc859bf4f-2m9ss             2/2       Running   0          13m
istio-policy-68797d879-ljm5t             2/2       Running   0          13m
istio-sidecar-injector-b88dfb954-6qclv   1/1       Running   0          13m
istio-telemetry-68787476f4-cb2rz         2/2       Running   0          13m
prometheus-76db5fddd5-d5mmq              1/1       Running   0          13m

[root@master istio-1.0.3]# cd ../istio-1.1.0-rc.1/install/kubernetes/helm/istio/charts/security/templates/
[root@master templates]# vim create-custom-resources-job.yaml
[root@master templates]# vim configmap.yaml
[root@master templates]# vim enable-mesh-mtls.yaml
[root@master templates]# vim enable-mesh-permissive.yaml

[root@master istio-1.1.0-rc.1]# helm upgrade istio install/kubernetes/helm/istio --namespace istio-system
Release "istio" has been upgraded. Happy Helming!
LAST DEPLOYED: Thu Feb 28 19:48:10 2019
NAMESPACE: istio-system
STATUS: DEPLOYED
...

[root@master istio-1.1.0-rc.1]# helm history istio
REVISION	UPDATED                 	STATUS    	CHART      	DESCRIPTION
1       	Thu Feb 28 19:33:28 2019	SUPERSEDED	istio-1.0.3	Install complete
2       	Thu Feb 28 19:48:10 2019	DEPLOYED  	istio-1.1.0	Upgrade complete
```

**Comment by morvencao** [comment]
Local test results:
```
root@morven1:~/istio-1.0.3# helm install --name istio --namespace istio-system install/kubernetes/helm/istio --tls
NAME:   istio
LAST DEPLOYED: Fri Mar  1 06:59:00 2019
NAMESPACE: istio-system
STATUS: DEPLOYED
...

root@morven1:~/istio-1.0.3# helm list --tls | grep istio
istio                 	1       	Fri Mar  1 06:59:00 2019	DEPLOYED	istio-1.0.3                 	istio-system
root@morven1:~/istio-1.0.3# kubectl -n istio-system get pod
NAME                                     READY     STATUS    RESTARTS   AGE
istio-citadel-78df8c67d9-t8p8s           1/1       Running   0          5m
istio-egressgateway-674686c846-r4dbj     1/1       Running   0          5m
istio-galley-58f566cb66-8vfgp            1/1       Running   0          5m
istio-ingressgateway-6bbdd58f8c-klznt    1/1       Running   0          5m
istio-pilot-7fc859bf4f-skwfs             2/2       Running   1          5m
istio-policy-68797d879-76db2             2/2       Running   0          2m
istio-policy-68797d879-7xxcj             2/2       Running   0          5m
istio-sidecar-injector-b88dfb954-pbpff   1/1       Running   0          5m
istio-telemetry-68787476f4-fhkdp         2/2       Running   0          5m
istio-telemetry-68787476f4-kvnk6         2/2       Running   0          1m
prometheus-76db5fddd5-95qfp              1/1       Running   0          5m
root@morven1:~# cd ../istio-1.1.0-rc.1/
root@morven1:~/istio-1.1.0-rc.1# helm upgrade istio-init --install ./install/kubernetes/helm/istio-init --namespace istio-system --tls
Release "istio-init" does not exist. Installing it now.
NAME:   istio-init
LAST DEPLOYED: Fri Mar  1 07:04:51 2019
NAMESPACE: istio-system
STATUS: DEPLOYED
...
root@morven1:~/istio-1.1.0-rc.1# kubectl -n istio-system get pod
NAME                                     READY     STATUS              RESTARTS   AGE
istio-citadel-78df8c67d9-t8p8s           1/1       Running             0          5m
istio-egressgateway-674686c846-r4dbj     1/1       Running             0          5m
istio-galley-58f566cb66-8vfgp            1/1       Running             0          5m
istio-ingressgateway-6bbdd58f8c-klznt    1/1       Running             0          5m
istio-ingressgateway-6bbdd58f8c-npktg    1/1       Running             0          16s
istio-init-crd-10-hzspv                  0/1       ContainerCreating   0          7s
istio-init-crd-11-r4c2p                  0/1       ContainerCreating   0          7s
istio-init-crd-certmanager-10-b2h6w      0/1       ContainerCreating   0          7s
istio-pilot-7fc859bf4f-skwfs             2/2       Running             1          5m
istio-policy-68797d879-76db2             2/2       Running             0          3m
istio-policy-68797d879-7xxcj             2/2       Running             0          5m
istio-sidecar-injector-b88dfb954-pbpff   1/1       Running             0          5m
istio-telemetry-68787476f4-fhkdp         2/2       Running             0          5m
istio-telemetry-68787476f4-kvnk6         2/2       Running             0          1m
prometheus-76db5fddd5-95qfp              1/1       Running             0          5m
root@morven1:~/istio-1.1.0-rc.1# helm list --tls | grep istio
istio                 	1       	Fri Mar  1 06:59:00 2019	DEPLOYED	istio-1.0.3                 	istio-system
istio-init            	1       	Fri Mar  1 07:04:51 2019	DEPLOYED	istio-init-1.1.0            	istio-system
root@morven1:~/istio-1.1.0-rc.1# helm upgrade istio ./install/kubernetes/helm/istio --namespace istio-system --tls
Release "istio" has been upgraded. Happy Helming!
LAST DEPLOYED: Fri Mar  1 07:06:23 2019
NAMESPACE: istio-system
STATUS: DEPLOYED
...
root@morven1:~/istio-1.1.0-rc.1# helm list --tls | grep istio
istio                 	2       	Fri Mar  1 07:06:23 2019	DEPLOYED	istio-1.1.0                 	istio-system
istio-init            	1       	Fri Mar  1 07:04:51 2019	DEPLOYED	istio-init-1.1.0            	istio-system
root@morven1:~/istio-1.1.0-rc.1# kubectl -n istio-system get pod
NAME                                      READY     STATUS      RESTARTS   AGE
istio-citadel-7959956db9-hk4tn            1/1       Running     0          3m
istio-galley-5c4ffb88db-c8wf5             1/1       Running     0          3m
istio-ingressgateway-69754b5f57-qz6kc     1/1       Running     0          3m
istio-init-crd-10-hzspv                   0/1       Completed   0          5m
istio-init-crd-11-r4c2p                   0/1       Completed   0          5m
istio-init-crd-certmanager-10-b2h6w       0/1       Completed   0          5m
istio-pilot-56f4c57d45-4k29c              2/2       Running     0          3m
istio-policy-fdddfc8cb-k25mv              2/2       Running     0          3m
istio-sidecar-injector-67b5777ffb-2sj5z   1/1       Running     0          3m
istio-telemetry-849fd966b-2xz9x           2/2       Running     0          3m
prometheus-89bc5668c-rhk4p                1/1       Running     0          3m
root@morven1:~/istio-1.1.0-rc.1# helm rollback istio 1 --tls
Rollback was a success! Happy Helming!
root@morven1:~/istio-1.1.0-rc.1# helm list --tls | grep istio
istio                 	3       	Fri Mar  1 07:10:24 2019	DEPLOYED	istio-1.0.3                 	istio-system
istio-init            	1       	Fri Mar  1 07:04:51 2019	DEPLOYED	istio-init-1.1.0            	istio-system
root@morven1:~/istio-1.1.0-rc.1# kubectl -n istio-system get pod
NAME                                     READY     STATUS      RESTARTS   AGE
istio-citadel-78df8c67d9-tq5vk           1/1       Running     0          26s
istio-egressgateway-674686c846-zcpfk     1/1       Running     0          28s
istio-galley-58f566cb66-hr6fw            1/1       Running     0          28s
istio-ingressgateway-6bbdd58f8c-pw6rd    1/1       Running     0          28s
istio-init-crd-10-hzspv                  0/1       Completed   0          6m
istio-init-crd-11-r4c2p                  0/1       Completed   0          6m
istio-init-crd-certmanager-10-b2h6w      0/1       Completed   0          6m
istio-pilot-7fc859bf4f-w7j5z             1/2       Running     0          28s
istio-policy-68797d879-mfrhs             2/2       Running     0          28s
istio-sidecar-injector-b88dfb954-5nx9j   1/1       Running     0          26s
istio-telemetry-68787476f4-dtzss         2/2       Running     0          28s
prometheus-76db5fddd5-nnmz5              1/1       Running     0          27s
```

**Comment by sdake** [comment]
@morvencao I did some preliminary investigation in this issue: https://github.com/istio/istio/issues/11990#issuecomment-468891090

**Comment by sdake** [comment]
PR works, personally tested, closing issue.

**Comment by clyang82** [comment]
I am so glad to see the issue was fixed in release-1.1. Thanks @morvencao & @sdake

**Comment by ostromart** [comment]
Verified the fix works for 1.0.5 -> release-1.1@fb449c9fa with Helm+tiller 2.12.3.
1. Started with clean cluster installed via tiller per 1.0 directions:
```
cd ~/istio-release/istio-1.0.5
kubectl apply -f install/kubernetes/helm/helm-service-account.yaml
￼￼￼helm init --service-account tiller
helm install install/kubernetes/helm/istio --name istio --namespace istio-system
```
--> Completed OK

2. Upgrade to 1.1 using release-1.1 branch:
```
# <git stuff>
git hist | head -n 5
fb449c9fa 2019-03-04| ...
1b077ea77 2019-03-05 | fix istio chart upgrade issues. (#12173) [Morven Cao] <-- Fix to test

helm install install/kubernetes/helm/istio-init --name istio-init --namespace istio-system
helm upgrade istio install/kubernetes/helm/istio --namespace istio-system
```
--> Completed OK

3. Wait a while, check running pods:
```
mostrowski@mostrowski-linux:~/go/src/istio.io/istio$ kc get pods -n istio-system
NAME                                      READY     STATUS      RESTARTS   AGE
istio-citadel-8c574d7c8-j9dz8             1/1       Running     0          7m
istio-galley-9d5986bc8-jtzwp              1/1       Running     0          7m
istio-ingressgateway-5b6d8f9ff8-469kx     1/1       Running     0          7m
istio-ingressgateway-5b6d8f9ff8-lqkh8     1/1       Running     0          7m
istio-init-crd-10-ppkzd                   0/1       Completed   0          8m
istio-init-crd-11-btq9c                   0/1       Completed   0          8m
istio-init-crd-certmanager-10-bx9gk       0/1       Completed   0          8m
istio-init-crd-certmanager-11-f8c8x       0/1       Completed   0          8m
istio-pilot-748f94dc68-ttztp              2/2       Running     0          7m
istio-sidecar-injector-57d48d8f6f-9w8tr   1/1       Running     0          7m
istio-telemetry-8656d64946-2xwfm          2/2       Running     2          7m
istio-telemetry-8656d64946-thzh7          2/2       Running     1          7m
prometheus-8647cf4bc7-5gnbt               1/1       Running     0          7m

kc get pods --namespace istio-system -o=jsonpath='{range .items[*]}{"\n"}{.metadata.name}{":\t"}{range .spec.containers[*]}{.image}{", "}{end}{end}\' |sort

istio-citadel-8c574d7c8-j9dz8:	gcr.io/istio-release/citadel:release-1.1-latest-daily, 
istio-galley-9d5986bc8-jtzwp:	gcr.io/istio-release/galley:release-1.1-latest-daily, 
istio-ingressgateway-5b6d8f9ff8-469kx:	gcr.io/istio-release/proxyv2:release-1.1-latest-daily, 
istio-ingressgateway-5b6d8f9ff8-lqkh8:	gcr.io/istio-release/proxyv2:release-1.1-latest-daily, 
istio-init-crd-10-ppkzd:	gcr.io/istio-release/kubectl:master-latest-daily, 
istio-init-crd-11-btq9c:	gcr.io/istio-release/kubectl:master-latest-daily, 
istio-init-crd-certmanager-10-bx9gk:	gcr.io/istio-release/kubectl:master-latest-daily, 
istio-init-crd-certmanager-11-f8c8x:	gcr.io/istio-release/kubectl:master-latest-daily, 
istio-pilot-748f94dc68-h78br:	gcr.io/istio-release/pilot:release-1.1-latest-daily, gcr.io/istio-release/proxyv2:release-1.1-latest-daily, 
istio-pilot-748f94dc68-ttztp:	gcr.io/istio-release/pilot:release-1.1-latest-daily, gcr.io/istio-release/proxyv2:release-1.1-latest-daily, 
istio-sidecar-injector-57d48d8f6f-9w8tr:	gcr.io/istio-release/sidecar_injector:release-1.1-latest-daily, 
istio-telemetry-8656d64946-2xwfm:	gcr.io/istio-release/mixer:release-1.1-latest-daily, gcr.io/istio-release/proxyv2:release-1.1-latest-daily, 
istio-telemetry-8656d64946-thzh7:	gcr.io/istio-release/mixer:release-1.1-latest-daily, gcr.io/istio-release/proxyv2:release-1.1-latest-daily, 
prometheus-8647cf4bc7-5gnbt:	docker.io/prom/prometheus:v2.3.1, \

```

**Comment by sdake** [comment]
agree, 3 people have tested and come to the same conclusion - this issue can be closed out.  Nice work @morvencao !
